core/src/test/java/org/elasticsearch/action/support/replication/TransportReplicationLimitTests.java

```diff
@@ -0,0 +1,220 @@
+        NetworkService networkService = new NetworkService(settings);
+
+        circuitBreakerService = new HierarchyCircuitBreakerService(settings, clusterSettings);
+        transportA = new NettyTransport(settings, threadPool, networkService, BigArrays.NON_RECYCLING_INSTANCE, Version.CURRENT, new
+            NamedWriteableRegistry(), circuitBreakerService);
+        transportServiceA = new TransportService(transportA, threadPool).start();
+        transportServiceA.acceptIncomingRequests();
+        final DiscoveryNode nodeA = new DiscoveryNode("nodeA", transportServiceA.boundAddress().publishAddress(),
+            emptyMap(), emptySet(), Version.CURRENT);
+
+        NettyTransport transportB = new NettyTransport(settings, threadPool, networkService, BigArrays.NON_RECYCLING_INSTANCE, Version
+            .CURRENT, new NamedWriteableRegistry(), circuitBreakerService);
+        transportServiceB = new TransportService(transportB, threadPool).start();
+        transportServiceB.acceptIncomingRequests();
+        nodeB = new DiscoveryNode("nodeB", transportServiceA.boundAddress().publishAddress(),
+            emptyMap(), emptySet(), Version.CURRENT);
+
+        transportA.connectToNode(nodeB);
+        clusterService = null;
+
+        clusterService = new ClusterService(settings, null, clusterSettings, threadPool, clusterName);
+        clusterService.setLocalNode(nodeA);
+        clusterService.setNodeConnectionsService(new NodeConnectionsService(Settings.EMPTY, null, null) {
+            @Override
+            public void connectToAddedNodes(ClusterChangedEvent event) {
+                // skip
+            }
+
+            @Override
+            public void disconnectFromRemovedNodes(ClusterChangedEvent event) {
+                // skip
+            }
+        });
+        clusterService.setClusterStatePublisher((event, ackListener) -> {
+        });
+        clusterService.start();
+    }
+
+    public void tearDown() throws Exception {
+        super.tearDown();
+        transportA.disconnectFromNode(nodeB);
+        transportServiceA.close();
+        transportServiceB.close();
+        clusterService.close();
+        terminate(threadPool);
+    }
+
+    public void testTransportReplicaActionNotSizeLimited() throws Exception {
+        final AtomicInteger count = new AtomicInteger(0);
+        final AtomicBoolean isRelocated = new AtomicBoolean(false);
+        String actionName = "testAction";
+        Action action = new Action(Settings.EMPTY, actionName, transportServiceA, clusterService, threadPool, count, isRelocated);
+
+        CountDownLatch replicaActionLatch = new CountDownLatch(1);
+        final AtomicReference<TransportException> capturedException = new AtomicReference<>();
+        final AtomicReference<TransportResponse> capturedResponse = new AtomicReference<>();
+        ShardId shard = new ShardId("test", "596D792A-0B84-11E6-9AF4-C738FDC8A409", 1);
```

Comment:
I always try to use realistic data in tests instead of "foo", "bla" or "_na_". ;) I looked in the code and it seems I am the only one doing it, so I'll use the more common "_na".